### DISCUSSION TITLE
Unable to connect via JDBC - DBeaver and Presto

### DISCUSSION THREAD

**Discussion by jmaxx37** [prestodb/presto#15697] [General]
Hi all,
I am on Presto version 0.245 and DBeaver version 7.3.4
It had the PrestoDB driver in it.

What do I put in for Database, Username and Pasword?
I just put in Database: system; User: presto, Passwd: presto and tried it.
Errored out:

Authentication using username/password requires SSL to be enabled

I am able to run it fine through the CLI.
Not sure how to connect through DBeaver.

**Comment by jmaxx37** [comment]
Bump

**Reply by bitsondatadev** [reply]
Hey @jmaxx37 👋

Brian here from [Trino, formerly known as PrestoSQL](https://trino.io/blog/2020/12/27/announcing-trino.html). Trino is similar to Presto but an improved query engine that the cofounders of Presto went off to create about 2 years back. With that, I think I can provide some guidance here.

First, assuming you've already set up authentication, you should make sure the you have the parameter SSL=true in your jdbc connect string.

If you haven't set up authentication I highly recommend you watch this video given by presto creator @dain https://trino.io/blog/2020/08/13/training-security.html.

Finally one note you should know that authentication in Presto has a [critical security vulnerability](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15087) that has existed in both Presto and Trino since the beginning but has only been addressed in Trino. If you plan to deploy to production I highly recommend looking at Trino. It has addressed this issue and added many features like an [Iceberg connector](https://trino.io/docs/current/connector/iceberg.html) and much more!

You can find me on [slack](https://trino.io/slack.html) if you have any questions!

**Reply by jmaxx37** [reply]
Hi @bitsondatadev !
I was finally able to connect via DBeaver.

Also, I went through this information. Seems super helpful...and lots more links to other informative material. 

That's awesome about Trino addressing the security vulnerability. I'll definitely look into it. 
I actually also have a Trino cluster setup and running, but Management is currently on Presto.

What would you say are other benefits of switching to Trino vs Presto?
Thanks!

**Reply by bitsondatadev** [reply]
So when discussing with management, it should be made clear that Trino is not a new technology. It's the same technology that shares the same codebase with PrestoDB since before 2019. The 4 original developers that started Presto and made it as successful as it is today, had some conflicts with the way Facebook wanted to manage the project, so they went off to make Trino. These projects largely do the same thing, and a lot of concepts remain the same. The main difference now is that Trino is only maintained by stakeholders for the project (i.e. the community) and doesn't favor any single entity. For more on this see the first rebranded Trino Community Broadcast I did with these guys for that story https://trino.io/episodes/8.html or read this blog, https://trino.io/blog/2020/12/27/announcing-trino.html. I believe understanding why the split happened helps management understand which project you should choose.

Regarding the features diffs, the easiest way to understand what has happened since Trino diverged from Presto, you can look at the list of features we put in our year-in-review blogs.

- https://trino.io/blog/2020/01/01/2019-summary.html
- https://trino.io/blog/2021/01/08/2020-review.html

A few highlights would be partition-based Dynamic Filtering (not just local DF), HDP3 support, Late Materialization, an Iceberg connector, Rubix Integration, various Support for SQL statements like OFFSET for example, Support for reading Hive views, and the list goes on...

Trino also has a strong community and as you may notice faster turnaround when you need help.

Again I recommend joining our slack if you want more details! :)

**Reply by aweisberg** [reply]
@bitsondatadev @jmaxx37 In Presto authorization for HTTP endpoints was added for https://github.com/prestodb/presto/issues/14639 and was released in [0.239](https://prestodb.io/docs/current/release/release-0.239.html).
